SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
+	// Step 4 - The server adds the new dynamic component
+	AddStep(
+		TEXT("DynamicSubobjectsTestServerAddComponent"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			// server has 1 more component than client (idk why)
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
 
-					if (IsValid(TestActor))
-					{
-						FinishStep();
-					}
-				}
-			},
-			StepTimeLimit);
-	}
+			// add new dynamic component to test actor
+			UStaticMeshComponent* AddedComponent = NewObject<UStaticMeshComponent>(TestActor, TEXT("ToRemoveComponent"));
+			AddedComponent->AttachToComponent(TestActor->GetRootComponent(), FAttachmentTransformRules::KeepWorldTransform);
+			AddedComponent->RegisterComponent();
+			AddedComponent->SetIsReplicated(true);
+
+			AssertEqual_Int(GetNumComponentsOnTestActor(), 4, TEXT("Now DynamicSubobjectTestActor should have 4 components"));
+			FinishStep();
+		},
+		nullptr);
+
+	// Step 5 - The client checks it sees the new component
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientSeeNewComponent"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			RequireEqual_Int(GetNumComponentsOnTestActor(), 4, TEXT("Now DynamicSubobjectTestActor should have 4 components"));
+			FinishStep();
+		},
+		StepTimeLimit);
+
+	for (int i = 0; i < DynamicComponentsPerClass + 2; ++i)
+	{
+		const bool LastStepLoop = i == DynamicComponentsPerClass + 1;
+
+		// Step 6 - Server moves the TestMovementCharacter of Client 1 to a remote location, so that it does not see the
+		// AReplicatedGASTestActor.
+		AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation)
+				&& RequireEqual_Vector(ClientOneSpawnedPawn->GetActorLocation(), CharacterRemoteLocation,
+									   TEXT("Client pawn was not moved to remote location"), 1.0f))
+			{
+				FinishStep();
+			}
+		});
 
-	{ // Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+		// Step 7 - Client 1 makes sure that the movement was correctly replicated
 		AddStep(
-			TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+			TEXT("DynamicSubobjectsTestClientCheckFirstMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
 			[this](float DeltaTime) {
-				ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
-				APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
-				ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
-				if (IsValid(PlayerController))
+				APawn* PlayerCharacter = GetFlowPawn();
+
+				if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should not be nullptr"))
+					&& RequireEqual_Vector(PlayerCharacter->GetActorLocation(), CharacterRemoteLocation,
```

Comment:
the require can be moved down into the if, I think

Reply:
done